Book-A-Desk.Api/BookingsHttpHandler.fs

```diff
@@ -100,6 +102,45 @@ module BookingsHttpHandler =
                     return! json response.Error next context
             }
 
+        let handleGetByEmailAndDateFromEventStore eventStore email date = asyncResult {
+            let (ReservationId aggregateId) = ReservationAggregate.Id
+            let! bookingEvents = eventStore.GetEvents aggregateId
+            return! ReservationsQueriesHandler.getUserBookingsStartFrom bookingEvents email date
+        }
+        
+        let handleGetByEmailAndDate () = fun next context ->
+            task {
+                let email = InputParser.parseEmailFromContext context
+                match email with
```

Comment:
I think there are better ways of writing this code instead of a cascading structure:
``` fsharp
match email, date with
| None, None ->
| Some email, None ->
| None, Some date ->
| Some email, Some date ->
```

I can't think of another way of writing this for now, but I think there also are better ways :), to avoid, multiple levels of indentation.

Reply:
Another way:
If InputParser returns an error, you can use the result computation expression to simplify every match:

``` fsharp
let! email = InputParser.parseEmailFromContext context
let! date = InputParser.parseEmailFromContext context
let eventStore = provideEventStore (context.GetService<IAmazonDynamoDB>())
let! bookings = handleGetByEmailAndDateFromEventStore eventStore email date
```